File: AccountantAndroid/app/src/main/java/com/kunkliricsi/accountant/model/entities/Expense.java
package com.kunkliricsi.accountant.model.entities;

import com.kunkliricsi.accountant.model.entities.enums.PayOption;

import java.io.PushbackReader;
import java.util.Date;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

@Entity(tableName = "expenses",
        foreignKeys = {
        @ForeignKey(
                entity = Report.class,
                parentColumns = "id",
                childColumns = "report"
        ),
        @ForeignKey(
                entity = Category.class,
                parentColumns = "id",
                childColumns = "category"
        ),
        @ForeignKey(
                entity = User.class,
                parentColumns = "id",
                childColumns = "purchaser"
        ),
})
public class Expense {

    @PrimaryKey
    public int id;

    @ColumnInfo(name = "amount")
    public int Amount;

    @ColumnInfo(name = "payoption")
    public int PayOption;

    @ColumnInfo(name = "dateofpurchase")
    public Date DateOfPurchase;

    @ColumnInfo(name = "report")
    public int ReportID;

    @ColumnInfo(name = "category")
    public int CategoryID;

    @ColumnInfo(name = "purchaser")
    public int PurchaserID;
}

File: src/web/Accountant.DAL/Entities/Expense.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace Accountant.DAL.Entities
{
    public class Expense
    {
        public int Id { get; set; }

        public int Amount { get; set; }

        public DateTime PurchaseDate { get; set; }

        public int ReportId { get; set; }
        public Report Report { get; set; }

        public int CategoryId { get; set; }
        public Category Category { get; set; }

        public int UserId { get; set; }
        public User User { get; set; }
    }
}
